packages/server/src/api/rest/index.ts

```diff
@@ -809,6 +829,89 @@ class RequestHandler extends APIHandlerBase {
         };
     }
 
+    private async processUpsert(
+        prisma: DbClientContract,
+        type: string,
+        _query: Record<string, string | string[]> | undefined,
+        requestBody: unknown,
+        modelMeta: ModelMeta,
+        zodSchemas?: ZodSchemas
+    ) {
+        const typeInfo = this.typeMap[type];
+        if (!typeInfo) {
+            return this.makeUnsupportedModelError(type);
+        }
+
+        const { error, attributes, relationships } = this.processRequestBody(type, requestBody, zodSchemas, 'create');
+
+        if (error) {
+            return error;
+        }
+
+        const matchFields = (requestBody as any).meta.matchFields;
```

Comment:
`matchFields` should always be `string[]`, right? Shall we zod validate it?

Reply:
Yes an optional array of strings. It makes sense to zod validate it. Where would this be added?

Reply:
I see you currently have a check `body?.meta?.operation === 'upsert'`. Maybe we can define a zod schema and replace that check with zod parse?

Reply:
I agree. Should the zod schema be a part of this file or be a part of the zodSchemas coming in to `handleRequest`? If the latter, where are these defined?

Reply:
Since it's only used locally I think we can just make it a member of the class. There's already one named `createUpdatePayloadSchema` defined 😄